daemon/src/maker_cfd.rs

```diff
@@ -418,6 +417,14 @@ impl Actor {
 
         Ok(())
     }
+
+    async fn handle_price_update(&mut self, quote: bitmex_price_feed::Quote) -> Result<()> {
+        self.cfd_feed.set_current_price(quote);
+
+        let mut conn = self.db.acquire().await?;
+        self.cfd_feed.update(&mut conn).await?;
```

Comment:
I think a cleaner way of doing this is to calculate the profit of every CFD here and then send a tuple of `(CFD, Profit)` to the cfd_feed.

The feed should be "dumb" and all the crucial calculation should be within the actor. Conversion the the actual feed model should also happen outside the actor though which is why I would just send over a tuple. Alternatively you could create another domain model that combines CFD + profit.

Reply:
What you described sounds a lot like the previous iteration of the PR, where the feed was "dumb" and I had the `CFD + profit` domain model. Wasn't that better in retrospect?

I struggle to see why the cfd actor needs to calculate the profit itself, as it does not make any use of it - it's just for the purpose of updating the feed. Doing it inside the CFD actors would mean:
a) need to store the current price (ie. quote) around, reducing encapsulation
b) need to duplicate that logic between 2 actors, as both maker and taker need to perform the calculations.

Reply:
In the previous iteration, the feed still did the calculation of the profit yeah?

At the moment, the `maker_cfd::Actor` is responsible for updating the feed. I just don't like that being "by-passed" by other actors. Are we really never going to need to current price for anything else inside the `maker_cfd::Actor`? It just seems worthwhile to centralize things within the `cfd::Actor`.

Reply:
The feed was just a watch channel - the calculation was done in the `to_sse_event` module, as the current quote was available there (it's managed by Rocket after all).

I can't see the need for the current price in cfd actors as for now (even in #108, the offer price will be calculated with the spread outside of the actor, and then `CfdNewOrderRequest` will take the offer price in).
 I have no idea whether it might be needed for CFD rollover or interacting with oracle? 

My initial motivation for keeping it out was that display of profit in the frontend is an implementation detail from the perspective of this Actor, whilst CfdActor takes care of things like ensuring the database records are up to date and new cfds are created/settled with the other party.